File: examples/lh-server/python/10-interrupts/workflow.py
from littlehorse.workflow import Workflow, WorkflowThread


class UnderpantsWorkflow:
    def __init__(self) -> None:
        self.all_underpants = None

    def wfLogic(self, wf: WorkflowThread) -> None:
        self.all_underpants = wf.declare_json_arr("all-underpants")
        wf.execute("start-underpants-collection")
        wf.wait_for_event("done-collecting-underpants")
        wf.execute("profit", self.all_underpants)
        wf.add_interrupt_handler("underpant-collected", self.handle_underpant)

    def handle_underpant(self, wf: WorkflowThread) -> None:
        interrupt_content = wf.declare_str("INPUT")
        self.all_underpants.assign(self.all_underpants.add(interrupt_content))
        wf.execute("collect-underpant", interrupt_content)

    def get_workflow(self) -> Workflow:
        return Workflow("collect-underpants", self.wfLogic)

File: examples/lh-server/go/10-interrupts/workflow.go
package main

import (
	"github.com/littlehorse-enterprises/littlehorse/sdk-go/lhproto"
	"github.com/littlehorse-enterprises/littlehorse/sdk-go/littlehorse"
)

var allUnderpants *littlehorse.WfRunVariable

func wfLogic(wf *littlehorse.WorkflowThread) {
	allUnderpants = wf.DeclareJsonArr("all-underpants")
	wf.Execute("start-underpants-collection")
	wf.WaitForEvent("done-collecting-underpants").RegisteredAsEmpty()
	wf.Execute("profit", allUnderpants)

	wf.HandleInterrupt("underpant-collected", handleUnderpant).RegisteredAs(lhproto.VariableType_STR)
}

func handleUnderpant(wf *littlehorse.WorkflowThread) {
	interruptContent := wf.DeclareStr("INPUT")
	allUnderpants.Assign(allUnderpants.Add(interruptContent))
	wf.Execute("collect-underpant", interruptContent)
}
